katdal/test/test_applycal.py

```diff
@@ -237,11 +237,20 @@ def test_left_right(self):
         assert_allclose(y[-1], 1j, rtol=1e-14)
 
 
-class TestGetCalProduct(object):
-    """Test the :func:`~katdal.applycal.get_cal_product` function."""
+class TestCalProductAccess(object):
+    """Test the :func:`~katdal.applycal.*_cal_product` functions."""
     def setup(self):
         self.cache = create_sensor_cache()
 
+    def test_has_cal_product(self):
+        assert_equal(has_cal_product(self.cache, ATTRS, 'K'), True)
+        assert_equal(has_cal_product(self.cache, ATTRS, 'B'), True)
+        assert_equal(has_cal_product(self.cache, ATTRS, 'G'), True)
+        assert_equal(has_cal_product(self.cache, ATTRS, 'haha'), False)
+        # Remove one part of multi-part cal product
+        self.cache.pop('cal_product_B0')
+        assert_equal(has_cal_product(self.cache, ATTRS, 'B'), False)
```

Comment:
If one part is missing (e.g. a cal process crashed), wouldn't we still want to apply what's left (and possibly interpolate over the missing part if it's not on the edge)?

Reply:
Good question. I'm currently pedantic in expecting all pieces to be there (which is what this unit test highlights :-)).